File: app/src/main/java/com/waray/spendhound/BorrowNowTransaction.java
package com.waray.spendhound;

public class BorrowNowTransaction {
    private String borrowId;
    private String borrowerID;
    private String lenderID;
    private String borrowerName;
    private String date;
    private String lender;
    private String borrowedAmountStr;
    private String status;
    private long timestamp;

    public BorrowNowTransaction() {
        // Default constructor required by Firebase Realtime Database
    }

    public BorrowNowTransaction(String borrowId, String borrowerID, String lenderID, String borrowerName, String date, String lender, String borrowedAmountStr, String status, long timestamp) {
        this.borrowId = borrowId;
        this.borrowerID = borrowerID;
        this.lenderID = lenderID;
        this.borrowerName = borrowerName;
        this.date = String.valueOf(date);
        this.lender = lender;
        this.borrowedAmountStr = borrowedAmountStr;
        this.status = status;
        this.timestamp = timestamp;
    }

    // Legacy constructor for backwards compatibility
    public BorrowNowTransaction(String borrowerID, String lenderID, String date, String borrowee, String borrowedAmountStr, String status) {
        this.borrowerID = borrowerID;
        this.lenderID = lenderID;
        this.date = String.valueOf(date);
        this.lender = borrowee;
        this.borrowedAmountStr = borrowedAmountStr;
        this.status = status;
        this.timestamp = System.currentTimeMillis();
    }

    public String getBorrowId() {
        return borrowId;
    }

    public void setBorrowId(String borrowId) {
        this.borrowId = borrowId;
    }

    public String getBorrowerID() {
        return borrowerID;
    }

    public void setBorrowerID(String borrowerID) {
        this.borrowerID = borrowerID;
    }

    public String getLenderID() {
        return lenderID;
    }

    public void setLenderID(String lenderID) {
        this.lenderID = lenderID;
    }

    public String getBorrowerName() {
        return borrowerName;
    }

    public void setBorrowerName(String borrowerName) {
        this.borrowerName = borrowerName;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getLender() {
        return lender;
    }

    public void setLender(String lender) {
        this.lender = lender;
    }

    public String getBorrowedAmountStr() {
        return borrowedAmountStr;
    }

    public void setBorrowedAmountStr(String borrowedAmountStr) {
        this.borrowedAmountStr = borrowedAmountStr;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/waray/spendhound/BorrowNowTransaction.kt
package com.waray.spendhound

import kotlinx.serialization.Serializable

@Serializable
class BorrowNowTransaction(
    var borrowId: String? = null,
    var borrowerID: String? = null,
    var lenderID: String? = null,
    var borrowerName: String? = null,
    var date: Long? = null,
    var lender: String? = null,
    var borrowedAmount: Double? = null,
    var status: String? = null,
    var timestamp: Long = System.currentTimeMillis(),
    var paymentSentDate: Long = 0,
    var monthYear: String? = null
) {
    constructor() : this(null, null, null, null, null, null, null, null, System.currentTimeMillis(), 0, null)

    // Getters for compatibility
    fun getBorrowId(): String? = borrowId
    fun getBorrowerID(): String? = borrowerID
    fun getLenderID(): String? = lenderID
    fun getBorrowerName(): String? = borrowerName
    fun getDate(): Long? = date
    fun getLender(): String? = lender
    fun getBorrowedAmount(): Double? = borrowedAmount
    fun getStatus(): String? = status
    fun getTimestamp(): Long = timestamp
    fun getPaymentSentDate(): Long = paymentSentDate
    fun getMonthYear(): String? = monthYear

    // Setters for compatibility
    fun setBorrowId(id: String?) { this.borrowId = id }
    fun setBorrowerID(id: String?) { this.borrowerID = id }
    fun setLenderID(id: String?) { this.lenderID = id }
    fun setBorrowerName(name: String?) { this.borrowerName = name }
    fun setDate(date: Long?) { this.date = date }
    fun setLender(lender: String?) { this.lender = lender }
    fun setBorrowedAmount(amount: Double?) { this.borrowedAmount = amount }
    fun setStatus(status: String?) { this.status = status }
    fun setTimestamp(timestamp: Long) { this.timestamp = timestamp }
    fun setPaymentSentDate(date: Long) { this.paymentSentDate = date }
    fun setMonthYear(monthYear: String?) { this.monthYear = monthYear }
}
